packages/@aws-cdk/aws-eks/lib/k8s-cluster-autoscaler.ts

```diff
@@ -0,0 +1,389 @@
+import autoscaling = require('@aws-cdk/aws-autoscaling');
+import iam = require('@aws-cdk/aws-iam');
+import core = require('@aws-cdk/core');
+import cdk = require('@aws-cdk/core');
+import { Cluster } from './cluster';
+import { KubernetesResource } from './k8s-resource';
+
+/**
+ * A Cluster Autoscaler
+ */
+export interface IKubernetesClusterAutoscaler extends core.IConstruct {
+  /**
+   *  The Cluster Autoscalers Cluster.
+   *
+   * [disable-awslint:ref-via-interface]
+   */
+  readonly cluster: Cluster;
+
+  /**
+   *  The IAM policy created by this construct.
+   */
+  readonly policy: iam.IPolicy;
+
+  /**
+   * The Kubernetes Resource that defines the Cluster Autoscaler K8s resources.
+   */
+  readonly clusterAutoscaler: KubernetesResource;
+}
+
+/**
+ * The properties for the Cluster Autoscaler.
+ */
+export interface KubernetesClusterAutoscalerProps {
+
+  /**
+   * The EKS cluster to deploy the cluster autoscaler to.
+   *
+   * [disable-awslint:ref-via-interface]
+   */
+  readonly cluster: Cluster;
+
+  /**
+   * An array of Autoscaling Groups, known as node groups, to configure for autoscaling.
+   *
+   * [disable-awslint:ref-via-interface]
+   */
+  readonly nodeGroups: autoscaling.AutoScalingGroup[];
+
+  /**
+   * The version of the Cluster Autoscaler to deploy.
+   *
+   * @default - the defualt version is v1.14.6
+   */
+  readonly version?: string;
+
+}
+
+/**
+ * The Cluster Autoscaler Construct. This will create a new IAM Policy, add labels to the ASGs, and
+ * deploy the Cluster Autoscaler manifest.
+ */
+export class KubernetesClusterAutoscaler extends cdk.Construct implements IKubernetesClusterAutoscaler {
+
+  /**
+   *  The cluster this autoscaler is assosciated with
+   *
+   * [disable-awslint:ref-via-interface]
+   */
+  public readonly cluster: Cluster;
+
+  /**
+   *  The IAM policy created by this construct.
+   */
+  public readonly policy: iam.IPolicy;
+
+  /**
+   * The Kubernetes Resource that defines the Cluster Autoscaler K8s resources.
+   */
+  public readonly clusterAutoscaler: KubernetesResource;
+
+  /**
+   * Constructs a new instance of the Cluster Autoscaler.
+   *
+   * @param scope cdk.Construct
+   * @param id string
+   * @param props ClusterAutoscalerProps
+   */
+  constructor(scope: cdk.Construct, id: string, props: KubernetesClusterAutoscalerProps) {
+    super(scope, id);
+
+    this.cluster = props.cluster;
+
+    const clusterName = props.cluster.clusterName;
+    const version = props.version === undefined ? 'v1.14.6' : props.version;
+
+    // define the cluster autoscaler policy statements
+    // https://docs.aws.amazon.com/en_pv/eks/latest/userguide/cluster-autoscaler.html#ca-create-ngs
+    const policyStatement = new iam.PolicyStatement();
+    policyStatement.addResources('*');
```

Comment:
Can we do better in terms of least privilege? I assume we know which ASGs and launch templates are used here, no?

Reply:
Making this more specific goes against our [docs](https://docs.aws.amazon.com/en_pv/eks/latest/userguide/cluster-autoscaler.html). I was trying to stick to the design the docs laid out.

Reply:
ok.